### PR TITLE
Amsterdam Email API + Alfresco Authentication plugin

### PR DESCRIPTION
**Pull request by haiko-ritense**
Toevoegen van Amsterdam Email API + Alfresco Authentication plugin

### CODE REVIEW

File: alfresco-authentication/backend/app/src/main/kotlin/com/ritense/valtimo/alfrescoauth/ValtimoApplication.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015-2022 Ritense BV, the Netherlands.
```
Comment (theo-ritense) [inline alfresco-authentication/backend/app/src/main/kotlin/com/ritense/valtimo/alfrescoauth/ValtimoApplication.kt:2]: Please update all copyright headers and make sure they've been added to every java or kotlin file.

File: alfresco-authentication/backend/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    id "org.jetbrains.kotlin.jvm" version "${kotlinVersion}"
+    id "org.jetbrains.kotlin.plugin.spring" version "${kotlinVersion}"
+    id "org.jetbrains.kotlin.plugin.allopen" version "${kotlinVersion}"
+    id "com.avast.gradle.docker-compose" version "0.14.9"
+}
+
+repositories {
+    mavenCentral()
+    mavenLocal()
+    jcenter()
```
Comment (theo-ritense) [inline alfresco-authentication/backend/app/build.gradle:26]: jcenter has been shutdown, please remove.

File: amsterdam-email-api/backend/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    id "org.jetbrains.kotlin.jvm" version "${kotlinVersion}"
+    id "org.jetbrains.kotlin.plugin.spring" version "${kotlinVersion}"
+    id "org.jetbrains.kotlin.plugin.allopen" version "${kotlinVersion}"
+    id "com.avast.gradle.docker-compose" version "0.14.9"
+}
+
+repositories {
+    mavenCentral()
+    mavenLocal()
+    jcenter()
```
Comment (theo-ritense) [inline amsterdam-email-api/backend/app/build.gradle:26]: please remove jcenter

File: amsterdam-email-api/backend/plugin/build.gradle [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+buildscript {
+    repositories {
+        mavenCentral()
+        jcenter()
```
Comment (theo-ritense) [inline amsterdam-email-api/backend/plugin/build.gradle:20]: please remove jcenter

File: amsterdam-email-api/backend/plugin/build.gradle [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    id "org.jetbrains.kotlin.jvm" version "${kotlinVersion}"
+    id "org.jetbrains.kotlin.plugin.spring" version "${kotlinVersion}"
+    id "org.jetbrains.kotlin.plugin.allopen" version "${kotlinVersion}"
+    id "com.avast.gradle.docker-compose" version "0.16.9"
+}
+
+repositories {
+    mavenCentral()
+    mavenLocal()
+    jcenter()
```
Comment (theo-ritense) [inline amsterdam-email-api/backend/plugin/build.gradle:42]: please remove jcenter

File: alfresco-authentication/backend/app/build.gradle [hunk 6]
```diff
@@ -0,0 +1,59 @@
+buildscript {
+    repositories {
+        mavenCentral()
+        jcenter()
```
Comment (theo-ritense) [inline alfresco-authentication/backend/app/build.gradle:4]: please remove jcenter

File: alfresco-authentication/backend/plugin/build.gradle [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+buildscript {
+    repositories {
+        mavenCentral()
+        jcenter()
```
Comment (theo-ritense) [inline alfresco-authentication/backend/plugin/build.gradle:20]: please remove jcenter

File: alfresco-authentication/backend/plugin/build.gradle [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    id "org.jetbrains.kotlin.jvm" version "${kotlinVersion}"
+    id "org.jetbrains.kotlin.plugin.spring" version "${kotlinVersion}"
+    id "org.jetbrains.kotlin.plugin.allopen" version "${kotlinVersion}"
+    id "com.avast.gradle.docker-compose" version "0.16.9"
+}
+
+repositories {
+    mavenCentral()
+    mavenLocal()
+    jcenter()
```
Comment (theo-ritense) [inline alfresco-authentication/backend/plugin/build.gradle:42]: please remove jcenter

File: amsterdam-email-api/backend/app/build.gradle [hunk 9]
```diff
@@ -0,0 +1,59 @@
+buildscript {
+    repositories {
+        mavenCentral()
+        jcenter()
```
Comment (theo-ritense) [inline amsterdam-email-api/backend/app/build.gradle:4]: please remove jcenter

**Review by theo-ritense — COMMENTED** [review]
Some initial comments (did not go over it all)

File: slack/frontend/projects/valtimo/slack/src/lib/components/post-message/post-message-configuration.component.html
Comment (Klaas-Ritense) [inline slack/frontend/projects/valtimo/slack/src/lib/components/post-message/post-message-configuration.component.html:1]: Please don't change the Slack plugin

File: slack/frontend/projects/valtimo/slack/src/lib/slack-plugin-module.ts
Comment (Klaas-Ritense) [inline slack/frontend/projects/valtimo/slack/src/lib/slack-plugin-module.ts:1]: Please don't change the Slack plugin

File: amsterdam-email-api/backend/plugin/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/client/EmailClient.kt [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.web.client.RestTemplate
+import java.net.URI
+
+
+private val logger = KotlinLogging.logger {}
+
+class EmailClient(
+    private val restTemplate: RestTemplate,
```
Comment (Klaas-Ritense) [inline amsterdam-email-api/backend/plugin/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/client/EmailClient.kt:17]: [RestTemplate is deprecated. Please use WebClient](https://medium.com/naukri-engineering/resttemplate-is-deprecated-use-webclient-a1dd7ca6394d)

File: amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin
Comment (Klaas-Ritense) [inline amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin:1]: Everything inside this folder `./gradle/` shouldn't be committed

File: amsterdam-email-api/backend/app/build/classes/kotlin/main/META-INF/app.kotlin_module
Comment (Klaas-Ritense) [inline amsterdam-email-api/backend/app/build/classes/kotlin/main/META-INF/app.kotlin_module:1]: Everything inside this `/build` folder shouldn't be committed.

File: amsterdam-email-api/backend/plugin/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/client/EmailClient.kt (hunk 10 above)
Comment (haiko-ritense) [reply amsterdam-email-api/backend/plugin/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/client/EmailClient.kt:17]: For compatibilty reasons in our project we have to use RestTemplate for now. Users can alter and use WebClient if their situation seems to fit.

File: alfresco-authentication/backend/gradle.properties [hunk 11]
```diff
@@ -0,0 +1,8 @@
+kotlinVersion=1.7.20
+springVersion=2.7.6
+springDependencyManagementVersion=1.0.11.RELEASE
+valtimoVersion=10.7.0.RELEASE
```
Comment (Klaas-Ritense) [inline alfresco-authentication/backend/gradle.properties:4]: It looks like the frontend part of this plugin uses a different version: `10.6.0`

Comment (haiko-ritense) [reply alfresco-authentication/backend/gradle.properties:4]: updated the frontend to 10.7.0

File: amsterdam-email-api/backend/app/build/classes/kotlin/main/META-INF/app.kotlin_module
Comment (haiko-ritense) [reply amsterdam-email-api/backend/app/build/classes/kotlin/main/META-INF/app.kotlin_module:1]: removed + add to gitignore

**Review by Klaas-Ritense — APPROVED** [review]

File: alfresco-authentication/backend/app/src/main/kotlin/com/ritense/valtimo/alfrescoauth/ApplicationReadyEventListener.kt [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+        val pluginConfig = pluginService.getPluginConfigurations(PluginConfigurationSearchParameters())
+            .firstOrNull { it.title == "Alfresco Auth configuration" }
+
+        if (pluginConfig != null) {
+            return pluginConfig
+        }
+
+        val configurationProperties = """
+            {
+                "clientId": "testClient",
```
Comment (theo-ritense) [inline alfresco-authentication/backend/app/src/main/kotlin/com/ritense/valtimo/alfrescoauth/ApplicationReadyEventListener.kt:60]: We should use externalised configuration for this. Putting credentials in code is not allowed by our development policy, so we should not put it there as an 'example' either.

File: amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin
Comment (Klaas-Ritense) [inline amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin:1]: This file shouldn't be committed

File: amsterdam-email-api/backend/app/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/ApplicationReadyEventListener.kt [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        val pluginConfig = pluginService.getPluginConfigurations(PluginConfigurationSearchParameters())
+            .firstOrNull { it.title == "Email API Amsterdam" }
+
+        if (pluginConfig != null) {
+            return pluginConfig
+        }
+
+        val configurationProperties = """
+            {   
+                "emailApiBaseUrl": "https://emailurl/mail/api/v1/mail",
```
Comment (theo-ritense) [inline amsterdam-email-api/backend/app/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/ApplicationReadyEventListener.kt:56]: Can't we use externalised configuration for this? Putting credentials and environment-specific configuration in code is not allowed by our policy, so we should not put it there as an 'example' either.

File: .gitignore [hunk 14]
```diff
@@ -166,3 +166,9 @@ Desktop.ini
 /slack/backend/app/build/
 /slack/backend/plugin/.gradle/
 /slack/backend/plugin/build/
+/amsterdam-email-api/backend/build/
```
Comment (theo-ritense) [inline .gitignore:169]: Can we add:
`**/.gradle` 
`**/build/`
`**/node_modules/`

instead?

File: alfresco-authentication/.gitignore
Comment (theo-ritense) [inline alfresco-authentication/.gitignore:1]: Is this needed? I'd prefer to only have one `.gitignore` in the root.

File: alfresco-authentication/frontend/projects/valtimo/alfresco/src/lib/components/alfresco-auth-configuration/alfresco-auth-configuration.component.scss
Comment (theo-ritense) [inline alfresco-authentication/frontend/projects/valtimo/alfresco/src/lib/components/alfresco-auth-configuration/alfresco-auth-configuration.component.scss:1]: Empty file?

File: alfresco-authentication/frontend/src/app/app.component.scss
Comment (theo-ritense) [inline alfresco-authentication/frontend/src/app/app.component.scss:1]: empty file? If it's really needed, put a copyright header in please.

File: alfresco-authentication/backend/plugin/src/main/kotlin/com/ritense/valtimo/alfresco/AlfrescoAuthenticationAutoConfiguration.kt [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+
+
```
Comment (theo-ritense) [inline alfresco-authentication/backend/plugin/src/main/kotlin/com/ritense/valtimo/alfresco/AlfrescoAuthenticationAutoConfiguration.kt:19]: Too many newlines

File: alfresco-authentication/backend/plugin/src/main/kotlin/com/ritense/valtimo/alfresco/AlfrescoAuthenticationPlugin.kt [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+
```
Comment (theo-ritense) [inline alfresco-authentication/backend/plugin/src/main/kotlin/com/ritense/valtimo/alfresco/AlfrescoAuthenticationPlugin.kt:18]: Too many newlines

File: alfresco-authentication/backend/plugin/src/main/kotlin/com/ritense/valtimo/alfresco/AlfrescoAuthenticationPluginFactory.kt [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+
```
Comment (theo-ritense) [inline alfresco-authentication/backend/plugin/src/main/kotlin/com/ritense/valtimo/alfresco/AlfrescoAuthenticationPluginFactory.kt:18]: Too many newlines

File: amsterdam-email-api/backend/app/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/ApplicationReadyEventListener.kt [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
```
Comment (theo-ritense) [inline amsterdam-email-api/backend/app/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/ApplicationReadyEventListener.kt:19]: Too many newlines

File: amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin
Comment (theo-ritense) [inline amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin:1]: Shouldn't this one be ignored/removed?

File: amsterdam-email-api/backend/plugin/docker-compose-integration-test.yml
Comment (theo-ritense) [inline amsterdam-email-api/backend/plugin/docker-compose-integration-test.yml:1]: empty file

File: amsterdam-email-api/backend/plugin/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/plugin/EmailApiAutoConfiguration.kt [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+    fun createEmailClient(restTemplate: RestTemplate, publisher: ApplicationEventPublisher): EmailClient {
+        return EmailClient(restTemplate, publisher)
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(RestTemplate::class)
+    fun createRestTemplate(): RestTemplate {
+        val httpClient: CloseableHttpClient =
+            HttpClients.custom()
+                .setSSLHostnameVerifier(NoopHostnameVerifier())
```
Comment (theo-ritense) [inline amsterdam-email-api/backend/plugin/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/plugin/EmailApiAutoConfiguration.kt:73]: Disabling SSL checks seem like a bad thing to do. Maybe this is acceptable in very specific cases, but certainly not for an application-wide RestTemplate.

File: alfresco-authentication/backend/plugin/docker-compose-integration-test.yml
Comment (theo-ritense) [inline alfresco-authentication/backend/plugin/docker-compose-integration-test.yml:1]: empty file

File: alfresco-authentication/frontend/src/assets/.gitkeep
Comment (theo-ritense) [inline alfresco-authentication/frontend/src/assets/.gitkeep:1]: empty file

File: amsterdam-email-api/frontend/projects/valtimo/amsterdam/emailapi/src/lib/components/amsterdam-emailapi-configuration/amsterdam-emailapi-configuration.component.scss
Comment (theo-ritense) [inline amsterdam-email-api/frontend/projects/valtimo/amsterdam/emailapi/src/lib/components/amsterdam-emailapi-configuration/amsterdam-emailapi-configuration.component.scss:1]: empty file

File: amsterdam-email-api/frontend/projects/valtimo/amsterdam/emailapi/src/lib/components/send-email/send-email-configuration.component.scss
Comment (theo-ritense) [inline amsterdam-email-api/frontend/projects/valtimo/amsterdam/emailapi/src/lib/components/send-email/send-email-configuration.component.scss:1]: empty file

File: amsterdam-email-api/frontend/src/app/app.component.scss
Comment (theo-ritense) [inline amsterdam-email-api/frontend/src/app/app.component.scss:1]: empty file

File: amsterdam-email-api/frontend/src/assets/.gitkeep
Comment (theo-ritense) [inline amsterdam-email-api/frontend/src/assets/.gitkeep:1]: empty file

File: amsterdam-email-api/frontend/src/styles.scss
Comment (theo-ritense) [inline amsterdam-email-api/frontend/src/styles.scss:1]: empty file

**Review by theo-ritense — COMMENTED** [review]
A lot of `.ts`, `.js` and `.kt` files are missing copyright headers still.

File: amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin
Comment (haiko-ritense) [reply amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin:1]: removed.

File: alfresco-authentication/backend/app/src/main/kotlin/com/ritense/valtimo/alfrescoauth/ApplicationReadyEventListener.kt (hunk 12 above)
Comment (haiko-ritense) [reply alfresco-authentication/backend/app/src/main/kotlin/com/ritense/valtimo/alfrescoauth/ApplicationReadyEventListener.kt:60]: Yes, I now use dummy values. Or would you recommend another value? But for sure, to work locally with it I will need a better solution.

File: amsterdam-email-api/backend/app/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/ApplicationReadyEventListener.kt (hunk 13 above)
Comment (haiko-ritense) [reply amsterdam-email-api/backend/app/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/ApplicationReadyEventListener.kt:56]: Yes, I now use dummy values. Or would you recommend another value? But for sure, to work locally with it I will need a better solution.

**Review by Klaas-Ritense — DISMISSED** [review]
.

File: alfresco-authentication/frontend/projects/valtimo/alfresco/src/lib/components/alfresco-auth-configuration/alfresco-auth-configuration.component.scss
Comment (haiko-ritense) [reply alfresco-authentication/frontend/projects/valtimo/alfresco/src/lib/components/alfresco-auth-configuration/alfresco-auth-configuration.component.scss:1]: conform Slack example

File: amsterdam-email-api/frontend/src/app/app.component.scss
Comment (haiko-ritense) [reply amsterdam-email-api/frontend/src/app/app.component.scss:1]: conform Slack example

File: .gitignore (hunk 14 above)
Comment (haiko-ritense) [reply .gitignore]: done

File: amsterdam-email-api/frontend/src/assets/.gitkeep
Comment (haiko-ritense) [reply amsterdam-email-api/frontend/src/assets/.gitkeep:1]: removed

File: alfresco-authentication/frontend/src/app/app.component.scss
Comment (haiko-ritense) [reply alfresco-authentication/frontend/src/app/app.component.scss:1]: done

File: alfresco-authentication/backend/plugin/src/main/kotlin/com/ritense/valtimo/alfresco/AlfrescoAuthenticationAutoConfiguration.kt (hunk 15 above)
Comment (haiko-ritense) [reply alfresco-authentication/backend/plugin/src/main/kotlin/com/ritense/valtimo/alfresco/AlfrescoAuthenticationAutoConfiguration.kt]: removed

File: amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin
Comment (haiko-ritense) [reply amsterdam-email-api/backend/plugin/.gradle/7.5/fileChanges/last-build.bin:1]: removed

File: amsterdam-email-api/backend/plugin/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/plugin/EmailApiAutoConfiguration.kt (hunk 19 above)
Comment (haiko-ritense) [reply amsterdam-email-api/backend/plugin/src/main/kotlin/com/ritense/valtimo/amsterdam/emailapi/plugin/EmailApiAutoConfiguration.kt:74]: I have add a comment to use it only in an internal network configuration. It also uses the    @ConditionalOnMissingBean(RestTemplate::class) statement. So a defalult application-wide RestTemplate should be picked up first.

**Review by theo-ritense — APPROVED** [review]

**Review by Klaas-Ritense — APPROVED** [review]

**Pull request merged**